### PR TITLE
[Unification] Move all encoders/transformers to models folders

### PR DESCRIPTION
**Pull request by RdoubleA**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #253
* #249
* #248
* #244
* #243
* #240
* __->__ #239

Differential Revision: [D38506859](https://our.internmc.facebook.com/intern/diff/D38506859)

## Summary
Move all model specific encoders and transformers from `modules/encoders/` and `modules/layers/transformer.py` to their corresponding model specific folders, following the naming convention `models/<model_name>/<modality>_encoder.py`. The encoders folder will remain for general use encoders.

## Test plan
`pytest -vv`

### CODE REVIEW

**Comment by sophiazhi** [comment]
there should be a change in the CLIP import statement [in videoclip](https://github.com/facebookresearch/multimodal/blob/main/examples/mugen/retrieval/video_clip.py#L14).

also maybe `test/models/albef/test_albef.py` should be renamed to `test/models/albef/test_model.py` (and similar change for clip and flava) since the model is now at `model.py`?

**Review by sophiazhi — CHANGES_REQUESTED** [review]
there should be a change in the CLIP import statement [in videoclip](https://github.com/facebookresearch/multimodal/blob/main/examples/mugen/retrieval/video_clip.py#L14).

also maybe `test/models/albef/test_albef.py` should be renamed to `test/models/albef/test_model.py` (and similar change for clip and flava) since the model is now at `model.py`?

File: test/models/clip/test_clip.py [hunk 1]
```diff
@@ -7,9 +7,9 @@
 import pytest
 import torch
 from test.test_utils import assert_expected, set_rng_seed
-from torchmultimodal.models.clip import CLIP
-from torchmultimodal.modules.encoders.clip_resnet_encoder import ResNetForCLIP
-from torchmultimodal.modules.encoders.clip_text_encoder import CLIPTextEncoder
+from torchmultimodal.models.clip.model import CLIP
+from torchmultimodal.models.clip.resnet_encoder import ResNetForCLIP
```
Comment (ebsmothers) [inline test/models/clip/test_clip.py:11]: Just wondering, do we wanna just call this image_encoder to be consistent with what we have for other models? (I know Ankita has a PR coming in for CLIP ViT encoder, so maybe we wanna keep things as you have them to keep these in separate files?)

**Review by ebsmothers — COMMENTED** [review]
Looks good overall. One question: why are the architecture tests getting moved around in this PR?

File: test/models/clip/test_clip.py (hunk 1 above)
Comment (RdoubleA) [reply test/models/clip/test_clip.py:11]: maybe we can place both in an `image_encoder.py` file for CLIP, what do you think? cc @ankitade

**Comment by RdoubleA** [comment]
> Looks good overall. One question: why are the architecture tests getting moved around in this PR?

those tests should've been moved already in an old PR but they weren't for some reason, so I just went ahead and did it in this PR

File: test/models/clip/test_clip.py (hunk 1 above)
Comment (ebsmothers) [reply test/models/clip/test_clip.py:11]: This sounds good to me. Ultimately will leave it up to Ankita as she is working more closely with the CLIP code these days

**Review by ebsmothers — APPROVED** [review]
Looks good to me pending the decision on the CLIP image encoders

**Comment by RdoubleA** [comment]
@RdoubleA has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38506859).

**Comment by RdoubleA** [comment]
@RdoubleA has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38506859).

File: test/models/clip/test_clip.py (hunk 1 above)
Comment (ankitade) [reply test/models/clip/test_clip.py]: yes image_encoder.py is fine

**Comment by RdoubleA** [comment]
@RdoubleA has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38506859).

**Comment by RdoubleA** [comment]
@RdoubleA has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38506859).

**Pull request closed without merging**